node/opr.go

```diff
@@ -1,33 +1,37 @@
 package node
 
 import (
+	"context"
 	"fmt"
 
 	"github.com/Factom-Asset-Tokens/factom"
 	"github.com/pegnet/pegnet/modules/grader"
+	log "github.com/sirupsen/logrus"
 )
 
-func (d *Pegnetd) Grade(block *factom.EBlock) (grader.GradedBlock, error) {
+func (d *Pegnetd) Grade(ctx context.Context, block *factom.EBlock) (grader.GradedBlock, error) {
 	if block == nil {
 		// TODO: Handle the case where there is no opr block.
 		// 		Must delay conversions if this happens
 		return nil, nil
 	}
 
-	if *block.ChainID != d.Tracking["opr"] {
+	if *block.ChainID != OPRChain {
 		return nil, fmt.Errorf("trying to grade a non-opr chain")
 	}
 
 	ver := uint8(1)
-	// TODO: Handle other than mainnet?
-	if block.Height > 210330 {
+	if block.Height >= 210330 {
 		ver = 2
 	}
 
 	var prevWinners []string = nil
-	prev := d.Pegnet.FetchPreviousBlock()
-	if prev != nil {
-		prevWinners = prev.WinnersShortHashes()
+	prev, err := d.Pegnet.SelectGrade(ctx, block.Height-1)
```

Comment:
A height could have no graded block. It is possible to skip a height, or even have an invalid eblock. We should really select the last valid graded block.

Reply:
my first thought was to just make it write a list of unbroken winners regardless of height (so instead of empty heights it would repeat entries) but i don't think that's necessary. changing it to a query that selects the most recent entry instead

Reply:
done